mindmeld/components/entity_resolver.py

```diff
@@ -42,60 +50,664 @@
+        else:
+            raise ValueError(f"Not supported model {model} to obtain number of layers")
+
+        return num_layers
+
+    @staticmethod
+    def _text_length(text):
+        """
+        Help function to get the length for the input text. Text can be either
+        a list of ints (which means a single text as input), or a tuple of list of ints
+        (representing several text inputs to the model).
+
+        Union[List[int], List[List[int]]]
+        """
+        if isinstance(text, dict):
+            return len(next(iter(text.values())))
+        elif len(text) == 0 or isinstance(text[0], int):
+            return len(text)
+        else:
+            return sum([len(t) for t in text])
+
+    @property
+    def device(self):
+        return "cuda" if _torch("is_available", sub="cuda") else "cpu"
+
+    @staticmethod
+    def get_hashid(config):
+        string = json.dumps(config, sort_keys=True)
+        return Hasher(algorithm="sha1").hash(string=string)
+
+    @staticmethod
+    def get_sentence_transformers_encoder(name_or_path,
+                                          output_type="mean",
+                                          all_modules=False,
+                                          quantize=True):
+        strans_models = _getattr("sentence_transformers.models")
+        strans = _getattr("sentence_transformers", "SentenceTransformer")
+
+        transformer_model = strans_models.Transformer(name_or_path,
+                                                      model_args={"output_hidden_states": True})
+        pooling_model = strans_models.Pooling(transformer_model.get_word_embedding_dimension(),
+                                              pooling_mode_cls_token=output_type == "cls",
+                                              pooling_mode_max_tokens=False,
+                                              pooling_mode_mean_tokens=output_type == "mean",
+                                              pooling_mode_mean_sqrt_len_tokens=False)
+        sbert_model = strans(modules=[transformer_model, pooling_model])
+
+        if quantize:
+            if not _is_module_available("torch"):
+                raise ImportError("`torch` library required to quantize models") from None
+
+            torch_qint8 = _getattr("torch", "qint8")
+            torch_nn_linear = _getattr("torch.nn", "Linear")
+            torch_quantize_dynamic = _getattr("torch.quantization", "quantize_dynamic")
+
+            transformer_model = torch_quantize_dynamic(
+                transformer_model, {torch_nn_linear}, dtype=torch_qint8
+            ) if transformer_model else None
+            pooling_model = torch_quantize_dynamic(
+                pooling_model, {torch_nn_linear}, dtype=torch_qint8
+            ) if pooling_model else None
+            sbert_model = torch_quantize_dynamic(
+                sbert_model, {torch_nn_linear}, dtype=torch_qint8
+            ) if sbert_model else None
+
+        if not all_modules:
```

Comment:
I think it would be better to just return None, None, sbert_model and use the positional index of the return tuple to find sbert. This way, the caller can to the special behavior of finding the index instead of making a varied behavior here in `get_sentence_transformers_encoder`